### PR TITLE
feat: add solo endorsements 

### LINKED ISSUE #40: [BE] User - Can have solo endorsement
### Description:
A user can have a solo endorsement granted to them. At this stage of development, this should be a simple relationship with a user and a position. By default, the endorsement should last 30 days.

A requirement from .NET is that the solo endorsements granted by a division should be visible somewhere on the website, so a query is required.

When a rating check fails for a user, the check should see if a Solo Endorsement has been granted to a user on the position the user is trying to book; ensuring they can still make bookings for a position they have a solo endorsement on, despite the rating check not being met.

### Acceptance Criteria / Requirements:
AC1: Solo endorsement can exist against a position/user
AC2: Solo endorsements last 30 days from the day they were granted
AC3: Bookings should be allowed when an endorsement is granted on the given position.
AC4: Should be able to query all active solo endorsements for audit purposes

### PR DESCRIPTION
**Pull request by AxonC**
Closes #40.

**Summary of Changes**

- Adds SoloEndorsement model w/ query scope for active endorsements
- Adds service to handle SoloEndorsement granting
- Adds query to retrieve all active solo endorsements as required by .NET
- Adds check to bookings to allow a user to book a position which they don't hold the rating for, but have an active solo endorsement

### CODE REVIEW

File: app/Modules/Endorsement/Solo/SoloEndorsementService.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+namespace App\Modules\Endorsement\Solo;
+
+use App\Modules\Position\Position;
+use App\User;
+use Carbon\Carbon;
+
+class SoloEndorsementService
+{
+    private const DEFAULT_DURATION_IN_DAYS = 30;
```
Comment (AndyTWF) [inline app/Modules/Endorsement/Solo/SoloEndorsementService.php:11]: Could make this a .env thing so it can be changed if needed?

**Review by AndyTWF — DISMISSED** [review]

File: app/Modules/Endorsement/Solo/SoloEndorsementService.php (hunk 1 above)
Comment (atoff) [reply app/Modules/Endorsement/Solo/SoloEndorsementService.php:11]: Personally, I think the current solution is ok? It's unlikely we would want in altered that often and there wouldn't be a reason for it to differ between environments?

File: tests/TestCase.php [hunk 2]
```diff
@@ -13,6 +13,6 @@ protected function setUp(): void
     {
         parent::setUp();
 
-        Carbon::setTestNow();
+        Carbon::setTestNow(Carbon::now());
```
Comment (atoff) [inline tests/TestCase.php]: This should still be without the `Carbon::now()` inside - this is meant to reset any carbon mocking between tests

**Review by atoff — CHANGES_REQUESTED** [review]

File: tests/TestCase.php (hunk 2 above)
Comment (AxonC) [reply tests/TestCase.php]: reverted

File: app/Modules/Endorsement/Solo/SoloEndorsementService.php (hunk 1 above)
Comment (AxonC) [reply app/Modules/Endorsement/Solo/SoloEndorsementService.php:11]: I'd keep it as is since it is business logic

**Review by atoff — APPROVED** [review]

**Pull request merged**